### PR TITLE
Update script and add boxploting

### LINKED ISSUE #1168: Update performance-check.sh
### File Location(s)

scripts/performance-check.sh

### Proposal

Improvements to #986 

- warm up executions
- perform the same count of measurements for both branches
- render results as box plots (python + pandas + seaborn)

### PR DESCRIPTION
**Pull request by gshep**
Resolves #1168 

- edit performance-check.sh: perform the same count of running for both branches
- render results with boxplots

@gear-tech/dev

### CODE REVIEW

**Comment by gshep** [comment]
Script will produce such easy to analyze 
<details><summary>box-ploted images</summary>

pallet-usage
State-rollback
</details>

**Review by breathx — CHANGES_REQUESTED** [review]
Also discussed. Optional flag to print python graphs is must have feature here

File: scripts/performance-check.sh [hunk 1]
```diff
@@ -1,44 +1,110 @@
...
+CURRENT_BRANCH=$3
 if [ -z "$CURRENT_BRANCH" ]; then
     CURRENT_BRANCH=`git branch --show-current`
 fi
 
-COUNT=$2
-if [ -z "$COUNT" ]; then
-    COUNT=100
-fi
+echo 'CURRENT_BRANCH = '$CURRENT_BRANCH' , COUNT = '$COUNT', ROOT_DIR = '$ROOT_DIR
```
Comment (breathx) [inline scripts/performance-check.sh:33]: dead code?
or append box plot also

File: scripts/performance-check.sh [hunk 2]
```diff
@@ -1,44 +1,110 @@
...
     mv $ROOT_DIR/target/runtime-test-junit.xml $ROOT_DIR/target/runtime-tests/$i
 done
 
-mkdir $ROOT_DIR/target/main_branch/
+rm -rf $ROOT_DIR/target/main_branch/
+mkdir -p $ROOT_DIR/target/main_branch/
 cargo run --package regression-analysis --release -- collect-data --data-folder-path $ROOT_DIR/target/tests/ --output-path $ROOT_DIR/target/main_branch/pallet-tests.json
 cargo run --package regression-analysis --release -- collect-data --disable-filter --data-folder-path $ROOT_DIR/target/runtime-tests/ --output-path $ROOT_DIR/target/main_branch/runtime-tests.json
 
 git checkout $CURRENT_BRANCH
```
Comment (breathx) [inline scripts/performance-check.sh:72]: seems here is some repeated code
can't we wrap it to fns?

File: scripts/performance-check.sh [hunk 3]
```diff
@@ -1,44 +1,110 @@
...
+    mv $ROOT_DIR/target/runtime-test-junit.xml $ROOT_DIR/target/runtime-tests/$i
+done
+
+rm -rf $ROOT_DIR/target/current_branch/
+mkdir -p $ROOT_DIR/target/current_branch/
+cargo run --package regression-analysis --release -- collect-data --data-folder-path $ROOT_DIR/target/tests/ --output-path $ROOT_DIR/target/current_branch/pallet-tests.json
+cargo run --package regression-analysis --release -- collect-data --disable-filter --data-folder-path $ROOT_DIR/target/runtime-tests/ --output-path $ROOT_DIR/target/current_branch/runtime-tests.json
+
+if [ "$BOXPLOT" != "0" ]; then
+    python3 $ROOT_DIR/scripts/performance-boxplot.py $ROOT_DIR/target/main_branch/pallet-tests.json $ROOT_DIR/target/current_branch/pallet-tests.json
```
Comment (breathx) [inline scripts/performance-check.sh:104]: and provide ability to just run box plot on pre-generated results

File: scripts/performance-check.sh [hunk 4]
```diff
@@ -1,44 +1,110 @@
...
+    $ROOT_DIR/scripts/gear.sh test rtest > $ROOT_DIR/target/runtime-tests-output/$i 2>&1
+done
+
+for i in `seq 1 $COUNT`; do
+    echo $i
+    $ROOT_DIR/scripts/gear.sh test rtest > $ROOT_DIR/target/runtime-tests-output/$i 2>&1
+    mv $ROOT_DIR/target/runtime-test-junit.xml $ROOT_DIR/target/runtime-tests/$i
+done
+
+rm -rf $ROOT_DIR/target/current_branch/
```
Comment (breathx) [inline scripts/performance-check.sh:98]: everywhere we use paths (especially with variables), wrap it into `""`, please

**Review by breathx — APPROVED** [review]

File: scripts/performance-check.sh (hunk 3 above)
Comment (gshep) [reply scripts/performance-check.sh:104]: I would like not to add additional logic to this wrapper script to keep it simple. In any case one can run this python-script by hand on pre-generated data.

**Pull request merged**